Sources/SwiftDocC/Utility/MarkupExtensions/ListItemExtractor.swift

```diff
@@ -710,3 +229,201 @@ struct TaggedListItemExtractor: MarkupRewriter {
         return nil
     }
 }
+
+// MARK: Extracting tags information
+
+/// Information about an extracted tag
+private struct ExtractedTag {
+    /// The raw name of the extracted tag
+    var rawTag: String
+    /// A known type of tag
+    var knownTag: KnownTag?
+    /// The range of the raw tag text
+    var tagRange: SourceRange?
+    /// The complete content related to this tag
+    var contents: [Markup]
+    /// The range of the tag and its content
+    var range: SourceRange?
+    
+    init(rawTag: String, tagRange: SourceRange?, contents: [Markup], range: SourceRange?) {
+        self.rawTag = rawTag
+        self.knownTag = .init(rawTag)
+        self.tagRange = tagRange
+        self.contents = contents
+        self.range = range
+    }
+    
+    enum KnownTag {
+        case returns
+        case `throws`
+        case parameter(String)
+        case parameters
+        
+        case dictionaryKey(String)
+        case dictionaryKeys
+        
+        case httpBody
+        case httpResponse(String)
+        case httpResponses
+        case httpParameter(String)
+        case httpParameters
+        case httpBodyParameter(String)
+        case httpBodyParameters
+        
+        init?(_ string: String) {
+            let components = string.components(separatedBy: .whitespaces).filter { !$0.isEmpty }
+            
+            switch components.first?.lowercased() {
+            case "returns":
+                self = .returns
+            case "throws":
+                self = .throws
+            case "parameter" where components.count == 2:
```

Comment:
Why is having 2 components special here?

Reply:
The first component is "parameter" and the second component is the parameter name. 

For example, ` - Parameter   someParameter : some description` would match this case with the  second component as "someParameter". 

However, ` - Parameter: some description` will not match this case because it doesn't specify which parameter the documentation applies to and ` - Parameter  one two: some description` doesn't match this case because it has 3 components. 

If we wanted we could change how we split the tag into components to only split once in order to be more permissive and support the "- parameter one two: " example as well. However, currently there's no supported language that allows whitespace in parameter names.